### DISCUSSION TITLE
Version not displayed in admin panel (or anywhere?)

### DISCUSSION THREAD

**Discussion by gh3597** [dani-garcia/vaultwarden#4936] [Q&A] [answered]
### Vaultwarden Support String

Clicking the 'generate support string' button does nothing so no support string is available. 

### Vaultwarden Build Version

1.32

### Deployment method

OS Package (apt, yum/dnf, pacman, apk, nix, ...)

### Custom deployment method

Installed as a DietPi native package using the DietPi software installer.

### Reverse Proxy

caddy, but accessing the admin page locally so this is irellevant to this issue.

### Host/Server Operating System

Linux

### Operating System Version

All versions (currently tested on DietPi v9.7.1/Debian 5.10.223-1

### Clients

Web Vault, CLI

### Steps To Reproduce

Log in to the admin pane.
Go to diagnostics
View the blank space where current version should be
![image](https://github.com/user-attachments/assets/b0b09180-b848-4fb1-bdb8-d66726bf6f65)

### Expected Result

See the installed version

### Actual Result

Not seeing the installed version

### Additional Context

I also can't see or find a way to display the version installed on the CLI either so I have no idea what server version I'm running so have to YOLO the updates as I don't know the relative changes.

**Comment by BlackDex** [comment]
I suggest to report this at the packager. It's not something we can resolve.

**Reply by gh3597** [reply]
So this is an issue for DietPi to fix? Assume that means this information is seen when installing VW on other OSes?

**Reply by BlackDex** [reply]
That depends, if git is used to pull the repo it should work, or if containers are used it will work. 
Or if packagers provide the right env variable during compile time it should also work.

**Reply by BlackDex** [reply, marked as answer]
For example, DietPi could use `VW_VERSION="1.32.7 - DietPi Edition"`

**Reply by gh3597** [reply]
I'll pass that info along then. Thanks.

**Reply by gh3597** [reply]
Fixed at DietPi. https://github.com/MichaIng/DietPi/issues/7364#issuecomment-2601925239

**Reply by MichaIng** [reply]
@BlackDex is it intended that "Server Latest" shows the very latest commit to the main branch, instead of the latest release/tag?

![image](https://github.com/user-attachments/assets/2cda9b9b-3041-4907-85fd-7419370b28cf)

**Reply by BlackDex** [reply]
It does that when it ends with a `-*`.
Maybe it can be a bit improved somehow.
But it isn't that big of an issue.

**Reply by MichaIng** [reply]
I see. I guess we can replace `-dietpi1` with `+dietpi1` instead.

**Reply by BlackDex** [reply]
That might be an option indeed. 
We use `(branch-name)` if the build is specifically build using a specific branch. So using a + i can make a difference between that too and just cut that part of during the version check.

Not sure though how that will work with other parts where we do version checks or have the availability to do that.

**Reply by MichaIng** [reply]
Ah, no need to change anything for us. @gh3597 just mentioned correctly that the (package) version suffix is actually not needed in the version string shown in web UI. We increment the suffix when doing e.g. a rebuild with new web vault (but same vaultwarden) version, or when changing something at configs, packaging, maintainer scripts or such. For the DEB package, the suffix is hence relevant, for the info on the web UI, where both, vaultwarden and web vault versions are separately listed, it is even somehow misleading, as some might interpret it like the vaultwarden sources where changed/patched. So our future builds set e.g. `VW_VERSION='1.33.2'` only, being the most correct value it can have.

And good to know for the future: So if we are really building from a different commit than the release tag, then we can add it with dash to `VW_VERSION`, to then correctly indicate the different source, being correctly interpreted by the web UI as well.

**Reply by BlackDex** [reply]
In don't mind looking into it. And adding something so users, and also if they need support, we, know and see where it comes from is only beneficial 

And, i might need to check some other places where we use the version, like css.

**Reply by MichaIng** [reply]
Okay just let me know. Alternatively, there might be also a dedicated variable/field to explicitly define who or how `vaultwarden` was built, if that is seen important. A bit similar to how some CLI tools' help output contain compiler and/or library versions they were compiled with resp. against. Otherwise I personally like the idea to have the version showing the release tag/version or `<version>-<commit>` only, hence the used source code version and in case commit.

**Reply by BlackDex** [reply]
That is possible by default, but then you need to use `git` including the git metadata. Without this data it's impossible to determine this automatically.
